RPG Wizardry/Assets/Scripts/Sorcery/Spells/Bookerang.cs

```diff
@@ -1,11 +1,147 @@
-﻿namespace nl.SWEG.RPGWizardry.Sorcery.Spells
+﻿using UnityEngine;
+using nl.SWEG.RPGWizardry.Avatar;
+using System.Collections;
+
+namespace nl.SWEG.RPGWizardry.Sorcery.Spells
 {
     public class Bookerang : Projectile
     {
+        #region Variables
+        /// <summary>
+        /// Bool for whether the book is moving forwards (false) or returning (true)
+        /// </summary>
+        private bool back = false;
+        /// <summary>
+        /// Bool to make the book pause in mid-air
+        /// </summary>
+        private bool pause = false;
+        /// <summary>
+        /// Float to keep track of how far we've moved
+        /// </summary>
+        private float movedSpace = 0;
+        /// <summary>
+        /// Sprite of the "crosshair" book
+        /// </summary>
+        private SpriteRenderer bookRenderer;
+        /// <summary>
+        /// Transform of the object that contains the sprite
+        /// (in a separate object so we can spin it independently)
+        /// </summary>
+        [SerializeField]
+        private Transform spriteTransform;
+        /// <summary>
+        /// Current location of the player, so we can return to it
+        /// </summary>
+        private Transform playerLocation;
+        /// <summary>
+        /// Position where we ended the forward movement
+        /// </summary>
+        private Vector3 savedPosition;
+        #endregion
+
+        #region Methods
+        #region Unity
+        /// <summary>
+        /// Get references for the player position and renderer of the "crosshair" book
+        /// </summary>
+        private void Start()
+        {
+            if (AvatarManager.Exists)
+            {
+                //Get the location of the player (to return to later)
+                playerLocation = AvatarManager.Instance.transform;
+                //Turn the "crosshair" book invisible so it looks like this projectile IS the book
+                bookRenderer = AvatarManager.Instance.BookRenderer;
+                bookRenderer.enabled = false;
+            }
+        }
+        #endregion
+        #region Protected
+        /// <summary>
+        /// Move in a straight line until we hit something or max out our range
+        /// Wait for a short time, then return to the player
+        /// </summary>
         protected override void Move()
         {
-            //turn around???
-            base.Move();
+            //SPIN TO WIN
+            spriteTransform.Rotate(Vector3.forward,1500 * Time.deltaTime,Space.World);
```

Comment:
Move rotationspeed to a (editor-)variable

Reply:
Sure